src/libstore/uds-remote-store.cc

```diff
@@ -2,16 +2,20 @@
 #include "unix-domain-socket.hh"
 #include "worker-protocol.hh"
 
+#include <cstring>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/socket.h>
-#include <sys/un.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
 
-#include <cstring>
-
+#ifdef _WIN32
+# include <winsock2.h>
```

Comment:
I liked the indentation to make nested `if..endif` easier to follow

Reply:
modern IDEs detect the _WIN32 switch an grey out others but so far didn't manage this in my visual studio code ssh IDE to pick up the compiler settings. for clion this required cmake to work properly and for other build systems at least one complete compile run and then IDEs sometimes pick up these settings.

this level of support would be nice for nix hacking!

Reply:
@qknight We have [instructions for the `clangd` LSP](https://nixos.org/manual/nix/unstable/contributing/hacking#editor-integration) if that's of use to you.
In support of this, we have `make compile_commands.json`. Maybe CLion supports that format too?

`bear` or similar isn't needed anymore since [recently](https://github.com/NixOS/nix/pull/10269).

Reply:
Also note that `clang-format` is configured to do this sort of indentation now.

Reply:
@Ericson2314 @roberth interesting! So i would have to use the stdenv.cc.isClang for formatting? Or would I install `nix-env -i clang-format` into my shell before `nix develop .#devShells.x86_64-linux.x86_64-w64-mingw32`?

      nativeBuildInputs = attrs.nativeBuildInputs or []
        # TODO: Remove the darwin check once
        # https://github.com/NixOS/nixpkgs/pull/291814 is available
         ++ lib.optional (stdenv.cc.isClang && !stdenv.buildPlatform.isDarwin) pkgs.buildPackages.bear
         ++ lib.optional (stdenv.cc.isClang && stdenv.hostPlatform == stdenv.buildPlatform) pkgs.buildPackages.clang-tools;

Tried to get my visual studio colde setup to work with `make compile_commands.json` but the problem is that I would need to make the login shell (it uses ssh to attach to the machine) have what the `nix develop ...` brings in and I couldn't make this happen. There is no field in the configuration, https://code.visualstudio.com/docs/remote/ssh, where one could set env vars or commands to run prior.

Reply:
Just run `make format`, it is in all shells.